payment-handler/can-make-payment-event-constructor.https.worker.js

```diff
@@ -0,0 +1,37 @@
+// https://w3c.github.io/payment-handler/#the-canmakepaymentevent
+
+'use strict';
+
+if (self.importScripts) {
+  importScripts('/resources/testharness.js');
+}
+
+test(() => {
+  try {
+    new CanMakePaymentEvent('test');
+  } catch (err) {
+    assert_unreached(`Unexpected exception: ${err.message}`);
+  }
+}, 'CanMakePaymentEvent can be constructed in service worker.');
+
+test(() => {
+  const ev = new CanMakePaymentEvent('test', {
+    bubbles: true,
+    cancelabel: true,
+    composed: true,
+  });
+  assert_false(ev.isTrusted, 'constructed in script, so not be trusted');
+  assert_true(ev.bubbles, 'set by EventInitDict');
+  assert_true(ev.cancelable, 'set by EventInitDict');
+  assert_true(ev.composed, 'set by EventInitDict');
+  assert_equals(ev.target, null, 'initially null');
+  assert_equals(ev.type, 'test');
+}, 'CanMakePaymentEvent can be constructed with an EventInitDict, even if not trusted');
+
```

Comment:
Sorry, there is one more set of tests missing here. You are testing `EventInit`, but not `CanMakePaymentEventInit`. I was expecting a test like:

```JS
test(() => {
  const ev = new CanMakePaymentEvent('test', {
    topLevelOrigin: "https://foo.com",
    paymentRequestOrigin: "https://bar.com",
    methodData: [],
    modifiers: [],
  });
  assert_false(ev.isTrusted, 'constructed in script, so not be trusted');
  assert_equals(ev.topLevelOrigin, "https://foo.com");
  assert_equals(ev.paymentRequestOrigin, "https://bar.com");
}, 'CanMakePaymentEvent can be constructed with an CanMakePaymentEventInit, even if not trusted');
```

And variations of the above, like providing weird `topLevelOrigin` values, and so on.